cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/LocalArtifactLoaderStage.java

```diff
@@ -87,8 +87,8 @@ public void process(AppDeploymentInfo deploymentInfo)
     String appVersion = deploymentInfo.getApplicationVersion();
     String configString = deploymentInfo.getConfigString();
 
-    NamespacedImpersonator classLoaderImpersonator =
-      new NamespacedImpersonator(artifactId.getParent(), impersonator);
+    EntityImpersonator classLoaderImpersonator =
+      new EntityImpersonator(artifactId.getParent(), impersonator);
```

Comment:
Should be calling getParent on artifact to get Namespace and hence get namespace owner ? This will spread the logic of whom we are impersonating for which entity all over the place. I think it will be better if we just pass artifact here and the OwnerAdmin will handle it. It's already responsible for giving the right owner to impersonate given an entity. This will keep the logic at one place.